trinity/sync/common/chain.py

```diff
@@ -124,6 +123,29 @@ def register_peer(self, peer: BasePeer) -> None:
                 else:
                     self.run_task(self.sync(peer))
 
+    @property
+    def _syncing(self) -> bool:
+        return self._peer_header_syncer is not None
+
+    @contextmanager
+    def _get_peer_header_syncer(
+            self, peer: HeaderRequestingPeer) -> Generator['PeerHeaderSyncer', None, None]:
```

Comment:
type-wise I think we figured out that we can do `Iterator[Thing]` here which is a bit cleaner, leaving of the dangly `None` bits.